### ISSUE TITLE
Spear reach as big as any other weapons reach

### ISSUE THREAD

**Issue by xXCubbie** [SilentChaos512/Silent-Gear#382]
## Versions

- **Silent Gear**: x.x.x  1.16.5 2.6.23
- **Silent's Gems**: x.x.x  installed, 1.16.3 3.7.14
- **Silent Lib**: x.x.x  1.16.3 4.9.6
- **Forge**: x.x.x 1.16.4 35.1.37
- **Modpack**: N/A 
- **Optifine Installed**:  yes   

## Expected Behavior
 I expected the spear to have a longer range

(Description here) I tested the hit range of various weapons

## Actual Behavior
 

(Description here) I made a cage around a creeper, went a few block away, until I couldn't hit the creeper with the spear anymore. Went one block closer to see if I could hit the creeper, which was possible. Now, to test if it actually has more reach, I tried hitting the creeper with a katana from the same maximal distance i could hit it using the spear. Turns out that not only katana, but also sword, axe and knife reach as far as the spear, even though the spear is supposed to reach farer

## Links/Images

- (Link) 

## Steps to Reproduce the Problem

1. Cage a creeper
2. Grab a spear and go away, test if you still can hit the creeper with each block you go
3. return to the block that is the farest away from the creeper but where the creeper is still in your spears reach
4. now try hitting him with other weapons, which should not be possible, since they were meant to have lower reach, but they still can reach as far as a spear

![Screenshot_18](https://user-images.githubusercontent.com/83208675/116995494-6a3a8480-acda-11eb-8353-4421027389cc.png)
thats my modlist

**Comment by Darkmega18** [comment]
I have a handful of these also and same issue (but a vastly larger modpack). Would you know if the spear is long or not in just the silent mods alone? perhaps give it a quick test? since you seem to have the vastly smaller modpack, Cubbie. :V 

If it suddenly works, I sorta want to point the finger to attribute fix potentially cause of it touching that kind of stuff...?

**Comment by SilentChaos512** [comment]
Working correctly in dev. Must be a mod conflict.

@xXCubbie Do the green crosshairs still appear when targeting a mod from beyond the vanilla attack range?

**Comment by Funnymeme1** [comment]
If it helps, I'm encountering the same problem. The green crosshair doesn't appear unless it's through a block, which is interesting. Trying to attack it through the block with the green indicator doesn't work.

**Comment by Darkmega18** [comment]
Tested with:
![image](https://user-images.githubusercontent.com/8104689/117087729-b09fea00-ad93-11eb-8e5e-e03b881350f1.png)

just those, installed via curseforge. literally just your mods. no spear working. and no green crosshair overlay either if thats meant to be a thing. wasn't even attribute fix, which is what I dropped in first to check for something bugging.

tested with a raw oak sword and spear. lined a zombie up infront of a tree with a fence around him. laid out a checker or blocks going back to check range. and swang sword at max range and moved back one. spear couldn't reach.
